hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/response/key/OMKeyRenameResponse.java

```diff
@@ -88,11 +88,22 @@ public void addToDBBatch(OMMetadataManager omMetadataManager,
           omMetadataManager.getOzoneKey(volumeName, bucketName, fromKeyName));
     } else if (createToKeyAndDeleteFromKey()) {
       // If both from and toKeyName are equal do nothing
-      omMetadataManager.getKeyTable().deleteWithBatch(batchOperation,
-          omMetadataManager.getOzoneKey(volumeName, bucketName, fromKeyName));
-      omMetadataManager.getKeyTable().putWithBatch(batchOperation,
-          omMetadataManager.getOzoneKey(volumeName, bucketName, toKeyName),
-          newKeyInfo);
+      if (!toKeyName.equals(fromKeyName)) {
+        omMetadataManager.getKeyTable().deleteWithBatch(batchOperation,
+            omMetadataManager.getOzoneKey(volumeName, bucketName, fromKeyName));
+        omMetadataManager.getKeyTable().putWithBatch(batchOperation,
+            omMetadataManager.getOzoneKey(volumeName, bucketName, toKeyName),
+            newKeyInfo);
+        // At this point we can also update the KeyIdTable.
+        if (newKeyInfo.getFileHandleInfo() != 0) {
```

Comment:
What does it mean for the fileHandleInfo to be zero? If it is zero, we do not delete the old Id from the keyIdTable?

Reply:
This means we never created the file handle for this key. The check is there to take care of this case. Eventually we should have file handle for all keys.

Reply:
Got it, thanks!